ggml/src/ggml-cuda/mmf.cuh

```diff
@@ -408,7 +445,11 @@ static __global__ void mul_mat_f_ids(
 #pragma unroll
                 for (int j0 = 0; j0 < tile_B::I; ++j0) {
                     const float2 tmp = vals_buf[curr_buf][j0];
+#if !defined(GGML_USE_HIP)
                     tile_xy[j0*tile_k_padded + threadIdx.x] = {tmp.x, tmp.y};
+#else
+                    tile_xy[j0*tile_k_padded + threadIdx.x] = ggml_cuda_cast<T, float2>(tmp);
+#endif // !defined(GGML_USE_HIP)
```

Comment:
Use `ggml_cuda_cast` unconditionally here and handle any vendor-specific instructions in `ggml_cuda_cast`.